### ISSUE TITLE
No improvement for cifar10-resnet example with Squeeze-and-Excitation operation on cifar10

### ISSUE THREAD

**Issue by KunpengLi1994** [tensorpack/tensorpack#1010]
Hi,

Thanks a lot for your great work. I'm trying to test performance of Squeeze-and-Excitation ResNet on cifar10. 

I use your code 'cifar10-resnet.py' under the folder of '/examples/ResNet/' and make some changes to line 52-72 about residual block based on your se_resnet_bottleneck code in 'resnet_model.py'.

The revised residual block is as follows:

        def residual(name, l, increase_dim=False, first=False):
            shape = l.get_shape().as_list()
            in_channel = shape[1]

            if increase_dim:
                out_channel = in_channel * 2
                stride1 = 2
            else:
                out_channel = in_channel
                stride1 = 1

            with tf.variable_scope(name):
                b1 = l if first else BNReLU(l)
                c1 = Conv2D('conv1', b1, out_channel, strides=stride1, activation=BNReLU)
                c2 = Conv2D('conv2', c1, out_channel)

                # add se-net
                squeeze = GlobalAvgPooling('gap', c2)
                squeeze = FullyConnected('fc1', squeeze, out_channel // 16, activation=tf.nn.relu)
                squeeze = FullyConnected('fc2', squeeze, out_channel, activation=tf.nn.sigmoid)
                data_format = get_arg_scope()['Conv2D']['data_format']
                ch_ax = 1 if data_format in ['NCHW', 'channels_first'] else 3
                shape = [-1, 1, 1, 1]
                shape[ch_ax] = out_channel
                c2 = c2 * tf.reshape(squeeze, shape)

                if increase_dim:
                    l = AvgPooling('pool', l, 2)
                    l = tf.pad(l, [[0, 0], [in_channel // 2, in_channel // 2], [0, 0], [0, 0]])

                l = c2 + l
                return l

Then I run following commend to test ResNet-110 with and without SE operation using two TITAN XP GPU, Tensorflow=1.3.0, py35, cudnn 6.0.  
`python cifar10-resnet.py -n 18`

The results for training error and validation error is as follows. Red line is SE-ResNet-110 and Blue line is ResNet-110. It seems the validation errors are both around 6%. There is no improvement as reported in the paper. Could please help to figure out the problem? Thanks.

![valid_error](https://user-images.githubusercontent.com/16794194/49902243-64a22280-fe31-11e8-9b70-1a538aaf8a68.png)

![train_error](https://user-images.githubusercontent.com/16794194/49902230-5bb15100-fe31-11e8-8bb6-1b58c1f7fb5c.png)

**Comment by ppwwyyxx** [comment]
Tensorpack issues are for questions/bug reports about the library. We do not answer general machine learning questions such as "why my model doesn't work".

From https://github.com/tensorpack/tensorpack/blob/master/.github/ISSUE_TEMPLATE.md:

> Some typical questions that we DO NOT answer:
"Could you improve/implement an example/paper ?" -- We have no plans to do so. We don't consider feature requests for examples or implement a paper for you. If you don't know how to do something yourself, you may ask a usage question.
"The examples do not perform well after I change the models/dataset/parameters/etc." Tensorpack maintainers make sure the examples perform well without modification. But it's your job to pick the model and parameters that are suitable for your own situation. We cannot help with such questions unless they appear to be a bug in tensorpack.
"Why my model doesn't work?", "I don't understand this paper you implement.", "How should I change the examples for my own dataset?" We do not answer machine learning questions.

**Comment by KunpengLi1994** [comment]
This issue is to test the effectiveness of your implemented Squeeze-and-Excitation operation in 'resnet_model.py'

  It seems has no improvement on your 'cifar10-resnet.py'.

**Comment by ppwwyyxx** [comment]
As quoted above:

> "The examples do not perform well after I change the models/dataset/parameters/etc." Tensorpack maintainers make sure the examples perform well without modification. But it's your job to pick the model and parameters that are suitable for your own situation. We cannot help with such questions unless they appear to be a bug in tensorpack.

The implemented SE is for training resnet on ImageNet and it is effective there.

**Comment by KunpengLi1994** [comment]
Sure, thanks
